.github/workflows/ci.yml

```diff
@@ -155,19 +155,15 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v5
-      - name: Setup Nightly Rust toolchain
-        uses: ./.github/actions/setup-builder
       - name: Install protoc
         uses: arduino/setup-protoc@v3
         with:
           repo-token: ${{ secrets.GITHUB_TOKEN }}
-      - name: Generate minimal versions lockfile
-        run: |
-          cargo generate-lockfile -Z direct-minimal-versions
+      - name: Setup Nightly Rust toolchain
+        uses: ./.github/actions/setup-builder
       - name: Setup MSRV Rust toolchain
         uses: ./.github/actions/setup-builder
         with:
           rust-version: ${{ env.rust_msrv }}
```

Comment:
needs both nightly and MSRV toolchain because 

`cargo generate-lockfile -Z direct-minimal-versions` requires nightly specified by `rust-toolchain.toml` 
`cargo +1.87 check --locked --workspace` then requires MSRV